### PR TITLE
Warnings on empty circuits, error when failure returned

### PR DESCRIPTION
**Pull request by albi3ro**

### CODE REVIEW

File: .readthedocs.yml [hunk 1]
```diff
@@ -1,3 +1,22 @@
...
+  - pip install -e git+https://github.com/XanaduAI/pennylane.git#egg=pennylane
+  - pip install -r requirements.txt
+  - pip install pytest pytest-cov wheel codecov --upgrade
+  - python3 setup.py bdist_wheel
+  - pip install dist/PennyLane_IonQ*.whl
+script:
+  - make coverage
+after_success:
+  - codecov
+=======
```
Comment (albi3ro) [inline .readthedocs.yml:19]:
```suggestion
```

File: pennylane_ionq/device.py [hunk 2]
```diff
@@ -114,6 +115,9 @@ def apply(self, operations, **kwargs):
         self.reset()
         rotations = kwargs.pop("rotations", [])
 
+        if len(operations) == 0 and len(rotations) == 0:
+            warnings.warn("Circuit is empty. Empty circuits return failures. Submitting anyway.")
+
```
Comment (Jaybsoni) [inline pennylane_ionq/device.py:120]: 👍🏼

File: tests/test_device.py [hunk 3]
```diff
@@ -100,6 +125,7 @@ def fake_response(self, resource_id=None):
         @qml.qnode(dev)
         def circuit():
             """Reference QNode"""
+            qml.PauliX(wires=0)
```
Comment (Jaybsoni) [inline tests/test_device.py:128]: was this the empty circuit causing the trouble ?

**Review by Jaybsoni — APPROVED** [review]
Looks good :)

**Pull request merged**